### PR TITLE
Created custom configurable/setting to ignore process names , (updated childProcessMonitor.ts and terminalConfiguration.ts) 

### LINKED ISSUE #131386: Terminal editor always marks as dirty
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 
```
Version: 1.59.1
Commit: 3866c3553be8b268c8a7f8c0482c0c0177aa8bfa
Date: 2021-08-19T11:53:52.479Z (3 days ago)
Electron: 13.1.7
Chrome: 91.0.4472.124
Node.js: 14.16.0
V8: 9.1.269.36-electron.0
OS: Darwin x64 20.5.0
```
- OS Version: macOS 11.4

### Steps to Reproduce:

1. Launch vscode with `code --disable-extensions`
2. Press `cmd+shift+p`, run `Terminal: Create Terminal in Editor Area`

I'm using zsh with the popular [**pure**](https://github.com/sindresorhus/pure) prompt, which will start a child `zsh` process. See more at: https://github.com/sindresorhus/pure/issues/215

It seems vscode already ignores some processes. Would it be a better way to provide a custom setting to ignore processes? 
https://github.com/microsoft/vscode/blob/3866c3553be8b268c8a7f8c0482c0c0177aa8bfa/src/vs/platform/terminal/node/childProcessMonitor.ts#L25-L31

### PR DESCRIPTION
**Pull request by san-coding**
This PR fixes #131386 

Added `zsh` to `ignoreProcessNames` array in `childProcessMonitor.ts` and created a new configurable/setting `terminal.integrated.confirmIgnoreProcesses` in  `terminalConfiguration.ts` , fixes  issue #131386 

### This PR resolves the issue that Background child processes created by `zsh` are not ignored by default , vscode already ignores some processes but a custom setting/configuration has been added to ignore processes.

- [x] I signed [CLA] [Screenshot 2021-08-25 at 6 15 31 PM ](https://cla.opensource.microsoft.com/microsoft/vscode?pullRequest=131486l)
- [x] I reproduced the issue on my machine
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"

### CODE REVIEW

**Comment by san-coding** [comment]
@Tyriar please review the pull request

**Comment by Tyriar** [comment]
@san-coding thanks for the change, we've just started testing v1.60 so we'll probably review this for v1.61 next week

**Comment by san-coding** [comment]
Thanks , will this be reviewed by the end of this month ?

**Comment by san-coding** [comment]
@Tyriar is there a need to add a dropdown (enum) for the newly created configurable/setting ?

**Review by Tyriar — CHANGES_REQUESTED** [review]
> is there a need to add a dropdown (enum) for the newly created configurable/setting ?

@san-coding since we don't know what the values could be (literally anything), we wouldn't be able to list them all out in an enum. Right now it's an array but I think you can tighten it to be a string array to make sure booleans, numbers, etc. don't get added.

File: src/vs/workbench/contrib/terminal/common/terminalConfiguration.ts [hunk 1]
```diff
@@ -455,6 +455,11 @@ const terminalConfiguration: IConfigurationNode = {
 			description: localize('terminal.integrated.customGlyphs', "Whether to draw custom glyphs for block element and box drawing characters instead of using the font, which typically yields better rendering with continuous lines. Note that this doesn't work with the DOM renderer"),
 			type: 'boolean',
 			default: true
+		},
+		[TerminalSettingId.ignoreProcessNames]: {
```
[the remark this answers is not in the document]
Comment (san-coding) [reply src/vs/workbench/contrib/terminal/common/terminalConfiguration.ts:459]: Ok thanks, I will make those changes, but do I have the roll back my existing commits ?

**Comment by san-coding** [comment]
@Tyriar I have changed `ignoreProcessNames` to `IgnoreProcessNames` in  `terminalConfiguration.ts`
 (Type rectified from last commit) and added `IgnoreProcessNames` definition to  `terminalPlatformConfiguration.ts`

**Review by Tyriar — CHANGES_REQUESTED** [review]
This won't compile as `ignoreProcessNames` was removed but wasn't replaced with using the setting yet. Please make sure you test the change and that it fixed the issue.

**Comment by san-coding** [comment]
> This won't compile as `ignoreProcessNames` was removed but wasn't replaced with using the setting yet. Please make sure you test the change and that it fixed the issue.

Sorry I don't understand what you meant by _`ignoreProcessNames` was removed but wasn't replaced with using the setting yet_ , we are using `IConfigurationService` in place of it right ?

**Comment by Tyriar** [comment]
> but I don't understand what you meant by ignoreProcessNames was removed but wasn't replaced with using the setting yet

You commented out the variable `ignoreProcessNames` but it's still being used and you didn't update the usage.

> we are using IConfigurationService in place of it right

You've injected the service but are not using it yet.

> could you please elaborate on why it won't compile

Please make sure the codebase compiles locally before sending PRs.

**Comment by san-coding** [comment]
> > but I don't understand what you meant by ignoreProcessNames was removed but wasn't replaced with using the setting yet
> 
> You commented out the variable `ignoreProcessNames` but it's still being used and you didn't update the usage.
> 
> > we are using IConfigurationService in place of it right
> 
> You've injected the service but are not using it yet.
> 
> > could you please elaborate on why it won't compile
> 
> Please make sure the codebase compiles locally before sending PRs.

Yes I feel the below code will cause an error, how can I fix this 
https://github.com/microsoft/vscode/blob/ee1655a82ebdfd38bf8792088a6602c69f7bbd94/src/vs/platform/terminal/node/childProcessMonitor.ts#L123

**Comment by san-coding** [comment]
> > > but I don't understand what you meant by ignoreProcessNames was removed but wasn't replaced with using the setting yet
> > 
> > 
> > You commented out the variable `ignoreProcessNames` but it's still being used and you didn't update the usage.
> > > we are using IConfigurationService in place of it right
> > 
> > 
> > You've injected the service but are not using it yet.
> > > could you please elaborate on why it won't compile
> > 
> > 
> > Please make sure the codebase compiles locally before sending PRs.
> 
> Yes I feel the below code will cause an error, how can I fix this , I don't see this being used anywhere else
> 
> https://github.com/microsoft/vscode/blob/ee1655a82ebdfd38bf8792088a6602c69f7bbd94/src/vs/platform/terminal/node/childProcessMonitor.ts#L123

**Comment by san-coding** [comment]
## I have un-commented `ignoreProcessNames`, will the issue be resolved now ?

**Comment by Tyriar** [comment]
> I have un-commented ignoreProcessNames, will the issue be resolved now ?

That will just do what it was doing before, you need to pull the setting in using the configuration service like this:

https://github.com/microsoft/vscode/blob/ee1655a82ebdfd38bf8792088a6602c69f7bbd94/src/vs/platform/terminal/node/terminalProfiles.ts#L42-L42

**Comment by san-coding** [comment]
> > I have un-commented ignoreProcessNames, will the issue be resolved now ?
> 
> That will just do what it was doing before, you need to pull the setting in using the configuration service like this:
> 
> https://github.com/microsoft/vscode/blob/ee1655a82ebdfd38bf8792088a6602c69f7bbd94/src/vs/platform/terminal/node/terminalProfiles.ts#L42-L42

Ok I will try to fix the issue, should I pull setting in `childProcessMonitor.ts` ?

**Comment by Tyriar** [comment]
Yep, in place of where ignoreProcessNames was being used

**Comment by san-coding** [comment]
> Yep, in place of where ignoreProcessNames was being used

Hey @Tyriar , I made the changes, but I feel that  they are incorrect and syntax is wrong , could you please verify and tell me what to do, thanks

**Comment by san-coding** [comment]
@Tyriar ??

**Comment by Tyriar** [comment]
This is still not really that close. It's imperative that you actually test the change when you contribute to OSS before you submit the pull request so as to not waste the time of the maintainers. Also it's bad etiquette to ping people on twitter and email, github is the place for that. Closing this to prevent more time from being wasted on both our parts.

**Comment by san-coding** [comment]
@Tyriar, I was just excited about my first pull request, wasn't receiving any response from your side, so I decided to ping you on Twitter and email you for help, I understand the inconvenience it might have caused, thank you for your assistance so far, I did learn a lot about VSCode codebase through this PR  and have learnt the lessons, won't repeat the same mistake, learning and contributing will continue 👍🏻

**Comment by Tyriar** [comment]
@san-coding glad you learned something and I don't intend to me mean, but it's taking too much time for me to guide you through this. I also just saw you pinged me on unrelated PRs, this is really quite annoying. Here's a screenshot of my inbox to give you an idea of what I'm working with:

image

PRs are always welcome but they must work, this either did compile or didn't fix the problem it claimed to (using the setting).

**Comment by san-coding** [comment]
> @san-coding glad you learned something and I don't intend to me mean, but it's taking too much time for me to guide you through this. I also just saw you pinged me on unrelated PRs, this is really quite annoying. Here's a screenshot of my inbox to give you an idea of what I'm working with:
> 
> image
> 
> PRs are always welcome but they must work, this either didn't compile or didn't fix the problem it claimed to (using the setting).

Oh I understand .. will do better next time

**Pull request closed without merging**